File: shared/src/commonMain/kotlin/io/github/jonathanpotts/cartographytable/shared/models/ServerModel.kt
package io.github.jonathanpotts.cartographytable.shared.models

import kotlinx.serialization.Serializable

/**
 * Stores data used to process a server.
 *
 * @property motd Message of the day on the server.
 * @property worlds Worlds on the server.
 */
@Serializable
data class ServerModel(
    var motd: String = "",
    var worlds: MutableList<WorldModel> = mutableListOf()
)

File: web/src/models/ServerModel.ts
import WorldModel from './WorldModel';

/**
 * Stores data used to process a server.
 */
interface ServerModel {
  /**
   * Message of the day on the server.
   */
  motd: string;

  /**
   * Worlds on the server.
   */
  worlds: Array<WorldModel>;
}

export default ServerModel;
